src/sentry/static/sentry/app/views/organizationStream/overview.jsx

```diff
@@ -122,7 +125,10 @@ const OrganizationStream = createReactClass({
     const newQuery = this.props.location.query;
 
     // Wait for saved searches to load before we attempt to fetch stream data
-    if (this.state.savedSearchLoading) {
+    if (this.props.savedSearchLoading) {
+      return;
+    } else if (prevProps.savedSearchLoading) {
+      this.fetchData();
```

Comment:
Could we hit this path multiple times before the issue list comes down? Should this also include `this.props.savedSearchLoading === false && prevProps.savedSearchLoading)`?

Reply:
Hmm that's what the `else if` checks -- I can make it more explicit if it makes it read more clearly.

Reply:
I'm a muppet, carry on.